site/app/controllers/student/SubmissionController.php

```diff
@@ -124,13 +124,22 @@ public function showHomeworkPage($gradeable_id, $gradeable_version = null) {
                     $can_inquiry = $this->core->getAccess()->canI("grading.electronic.grade_inquiry", ['graded_gradeable' => $graded_gradeable]);
                 }
 
+                $id = $this->core->getUser()->getId();
+
+                if($gradeable->isTeamAssignment()){
```

Comment:
You're already doing the checking here, so just remove the else from the if statement, and rename $teamID to $id where you set it inside the if statement. This will overwrite the user id to the team ID only if it is a team assignment which is what the view is doing (The logic should be done in the controller). Then you can remove the duplicated logic from the view. This will also simplify the code and make it easier to see what is going on.